ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/PagerDutyIntegrationForm.tsx

```diff
@@ -62,6 +62,12 @@ function PagerDutyIntegrationForm({
     const [isUpdatingStoredCredential, setIsUpdatingCredential] = useState(false);
     const isStoredCredentialInputEnabled = isCreating || isUpdatingStoredCredential;
 
+    let formInitialValues = structuredClone(defaultValues);
+    if (initialValues) {
+        merge(formInitialValues, initialValues);
+        formInitialValues = clearStoredCredentials(formInitialValues, storedCredentialKeyPaths);
```

Comment:
As independent eyes, you can evaluate and decide whether explicit assignment prevents misunderstanding.

Although helper function mutates `formInitialValues` the assignment leaves no doubt about the data flow.
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Integrations/utils/integrationUtils.ts#L60-L77

Reply:
The only example of update stored credentials in **component** state instead of **form** state.